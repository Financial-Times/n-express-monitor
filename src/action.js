import { logAction, compose } from '@financial-times/n-auto-logger';
import { metricsAction, tagService } from '@financial-times/n-auto-metrics';

import { getConfig } from './setup';

const noDecoration = inputFunction => inputFunction;

export const monitorService = (serviceName, functionBundle) => {
	const { metrics } = getConfig();
	return compose(
		tagService(serviceName),
		metrics ? metricsAction : noDecoration,
		logAction,
	)(functionBundle);
};

export const monitorModule = functionBundle => {
	const { metrics } = getConfig();
	return compose(
		metrics ? metricsAction : noDecoration,
		logAction,
	)(functionBundle);
};
