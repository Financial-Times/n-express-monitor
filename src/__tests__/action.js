import * as nAutoMetrics from '@financial-times/n-auto-metrics';

import setupMonitor from '../setup';
import { monitorService, monitorModule } from '../action';

jest.mock('@financial-times/n-auto-metrics', () => ({
	tagService: () => inputFunction => inputFunction,
	metricsAction: jest.fn(inputFunction => inputFunction),
	initAutoMetrics: jest.fn(),
}));

describe('monitorService', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	const app = {
		use: jest.fn(),
	};

	const metrics = {};

	it('disable metricsAction if metrics instance is not set', () => {
		setupMonitor({ app });
		const serviceActions = {
			method: () => {},
		};
		monitorService('mock-service', serviceActions);
		expect(nAutoMetrics.metricsAction.mock.calls).toMatchSnapshot();
	});

	it('enable metricsAction if metrics instance is set', () => {
		setupMonitor({ app, metrics });
		const serviceActions = {
			method: () => {},
		};
		monitorService('mock-service', serviceActions);
		expect(nAutoMetrics.metricsAction.mock.calls).toMatchSnapshot();
	});
});

describe('monitorModule', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	const app = {
		use: jest.fn(),
	};

	const metrics = {};

	it('disable metricsAction if metrics instance is not set', () => {
		setupMonitor({ app });
		const moduleFunctions = {
			method: () => {},
		};
		monitorModule(moduleFunctions);
		expect(nAutoMetrics.metricsAction.mock.calls).toMatchSnapshot();
	});

	it('enable metricsAction if metrics instance is set', () => {
		setupMonitor({ app, metrics });
		const moduleFunctions = {
			method: () => {},
		};
		monitorModule(moduleFunctions);
		expect(nAutoMetrics.metricsAction.mock.calls).toMatchSnapshot();
	});
});
