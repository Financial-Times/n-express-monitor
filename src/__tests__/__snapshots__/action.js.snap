// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monitorModule disable metricsAction if metrics instance is not set 1`] = `Array []`;

exports[`monitorModule enable metricsAction if metrics instance is set 1`] = `
Array [
  Array [
    Object {
      "method": [Function],
    },
  ],
]
`;

exports[`monitorService disable metricsAction if metrics instance is not set 1`] = `Array []`;

exports[`monitorService enable metricsAction if metrics instance is set 1`] = `
Array [
  Array [
    Object {
      "method": [Function],
    },
  ],
]
`;
